scargo/commands/build.py

```diff
@@ -17,76 +17,92 @@
 logger = get_logger()
 
 
-def scargo_build(profile: str, target: Optional[ScargoTarget]) -> None:
+def scargo_build(
+    profile: str, target: Optional[ScargoTarget], all_targets: bool = False
+) -> None:
     """
     Build project exec file.
 
     :param str profile: Profile
+    :param target: Target to build
+    :param bool all_targets: Build all targets
     :return: None
     """
     config = prepare_config()
-    build_target = get_target_or_default(config, target)
-
     project_dir = config.project_root
-    if not project_dir:
-        logger.error("Current working directory is not part of scargo project.")
-        sys.exit(1)
 
     if not Path(project_dir, "CMakeLists.txt").exists():
         logger.error("File `CMakeLists.txt` does not exist.")
         logger.info("Did you run `scargo update`?")
         sys.exit(1)
 
-    logger.info(f"Running scargo build for {build_target.id} target")
     conan_add_default_profile_if_missing()
     conan_add_remote(project_dir, config)
     conan_source(project_dir)
 
-    build_dir = Path(project_dir, build_target.get_profile_build_dir(profile))
-    build_dir.mkdir(parents=True, exist_ok=True)
-    profile_name = build_target.get_conan_profile_name(profile)
-
-    try:
-        subprocess.run(
-            [
-                "conan",
-                "install",
-                ".",
-                "-pr",
-                f"./config/conan/profiles/{profile_name}",
-                "-of",
-                build_dir,
-                "-b",
-                "missing",
-            ],
-            cwd=project_dir,
-            check=True,
-        )
-        subprocess.run(
-            [
-                "conan",
-                "build",
-                ".",
-                "-pr",
-                f"./config/conan/profiles/{profile_name}",
-                "-of",
-                build_dir,
-            ],
-            cwd=project_dir,
-            check=True,
-        )
-
-        get_logger().info("Copying artifacts...")
-        # This is a workaround so that different profiles can work together with conan
-        # Conan always calls CMake with '
-        subprocess.run(
-            f"cp -r -l -f {build_dir}/build/{config.profiles[profile].cmake_build_type}/* .",
-            cwd=build_dir,
-            shell=True,
-            check=True,
-        )
-        get_logger().info("Artifacts copied")
-
-    except subprocess.CalledProcessError:
-        logger.error("Scargo build failed")
-        sys.exit(1)
+    if not all_targets:
+        _scargo_build_targets(config, profile, [get_target_or_default(config, target)])
```

Comment:
this else is really needed? if you will have single target I will handle single list element only, so always handle the list of targets, eventually with only one element

Reply:
It's because for multitarget projects you can also specify for which target you want to build if you don't want to build for all targets. Also, we build only the default (first) target by default.